src/classes/ADDR_Validator_TDTM.cls

```diff
@@ -58,7 +60,8 @@ public class ADDR_Validator_TDTM extends TDTM_Runnable {
         
         if(!alreadyRun) {
 	        Addr_Verification_Settings__c settings = UTIL_CustomSettingsFacade.getAddressVerificationSettings();
-	        if(trigger.isBefore && settings.enable_automatic_verification__c) {        	
+	        if ((triggerAction == TDTM_Runnable.Action.BeforeInsert || triggerAction == TDTM_Runnable.Action.BeforeUpdate) &&
```

Comment:
trigger is only valid if the trigger handler is called syncronously.  when i was testing async, you would get an exception looking at trigger since it doesn't exist in the future call!